x-pack/plugins/fleet/server/services/agents/crud.ts

```diff
@@ -120,6 +131,17 @@ export async function getAgentsByKuery(
     filters.push(ACTIVE_AGENT_CONDITION);
   }
 
+  if (showManaged === false) {
```

Comment:
this is the main change, to query agents for non-managed policies only
unfortunately I needed to add `soClient` for usages everywhere